server/hap2/test/TestHaplib.py

```diff
@@ -210,3 +213,156 @@ def test_notifiy(self):
+
+    def test_parse_received_message_invalid_error_reply(self):
+        test_message = '{"error": {"code": 1, "message": "test_message"},"id": 1}'
+        result = haplib.Utils.parse_received_message(test_message, None)
+        self.assertEquals("Invalid error message: " + test_message,
+                          str(result.error_message))
+        self.assertEquals(1, result.message_id)
+
+    def test_convert_json_to_dict_success(self):
+        test_json_string = '{"test_key":"test_value"}'
+
+        exact_result = json.loads(test_json_string)
+        unnesessary_result, result = \
+                haplib.Utils._convert_json_to_dict(test_json_string)
+        self.assertEquals(result, exact_result)
+
+    def test_convert_json_to_dict_failure(self):
+        test_json_string = '{"test_key": test_value}'
+
+        exact_result = (-32700, None)
+        result = haplib.Utils._convert_json_to_dict(test_json_string)
+        self.assertEquals(exact_result, result)
+
+    def test_check_error_dict_success(self):
+        error_dict = {"error": {"code": 1, "message": "test_message", "data": "test_data"},"id": 1}
+        common.assertNotRaises(haplib.Utils._check_error_dict, error_dict)
+
+    def test_check_error_dict_failure(self):
+        error_dict = {"error": {"code": 1, "message": "test_message"},"id": 1}
+        try:
+            haplib.Utils._check_error_dict(error_dict)
+            raise
+        except:
+            pass
+
+    def test_is_allowed_procedure_success(self):
+        result = haplib.Utils.is_allowed_procedure("exchangeProfile",
+                                                   ["exchangeProfile"])
+        self.assertIsNone(result)
+
+    def test_is_allowed_procedure_failure(self):
+        result = haplib.Utils.is_allowed_procedure("test_procedure_name",
+                                                   ["exchangeProfile"])
+        self.assertEquals(result, -32601)
+
+    def test_validate_arguments_success(self):
+        test_json_string = '{"method": "exchangeProfile", "params": {"name":"test_name", "procedures":["exchangeProfile"]}}'
+        test_json_dict = json.loads(test_json_string)
+        result = haplib.Utils.validate_arguments(test_json_dict)
+        self.assertTrue((None, None), result)
+
+    def test_validate_arguments_fail(self):
+        test_json_string = '{"method": "exchangeProfile", "params": {"name":"test_name", "procedures":"exchangeProfile"}}'
+        test_json_dict = json.loads(test_json_string)
+        result = haplib.Utils.validate_arguments(test_json_dict)
+        self.assertEquals(result[0], -32602)
+
+    def test_generate_request_id(self):
+        result = haplib.Utils.generate_request_id(0x01)
+
+        self.assertTrue(0 <= result <= 1111111111111)
+
+    def test_translate_unix_time_to_hatohol_time(self):
+        result = haplib.Utils.translate_unix_time_to_hatohol_time(0)
+        self.assertEquals(result, "19700101090000.000000")
```

Comment:
translate_unix_time_to_hatohol_time is received utc time from hatohol server.
So, 0 is utc time.
I think, translate_hatohol_time_to_unix_time need to think about utc time.
But, translate_unix_time_to_hatohol_time not need think about utc time.